crates/polars-plan/src/plans/conversion/dsl_to_ir/expr_to_ir.rs

```diff
@@ -356,6 +356,7 @@ pub(super) fn to_aexpr_impl(
                                 dtype: DataType::Categorical(_, _) | DataType::Enum(_, _),
                                 ..
                             } => {
+                                // TODO @ cat-rework: why not?
```

Comment:
Nesting casting is of categorical does not work, but this was probably for another reason

Reply:
https://github.com/pola-rs/polars/pull/23432